draft-ietf-mls-protocol.md

```diff
@@ -1848,9 +1848,14 @@ enum {
     (255)
 } SenderType;
 
+struct {
+    opaque sender_identity;
+    opaque sender_endpoint_id;
+} Client
+
 struct {
     SenderType sender_type;
-    uint32 sender;
+    Client sender;
```

Comment:
I don't think this quite works.  Remember that we have multiple sender types to accommodate: `member`, `preconfigured`, and `new_member`.  This covers the `member` case.  We should probably have a switch/variant struct here (probably always should have).

```
struct {
    SenderType sender_type;
    switch (sender_type) {
        case member:        Client client;
        case preconfigured: opaque key_id<0..255>;
        case new_member:    struct{};
    }
} Sender;
```

Reply:
Note that you can still use `Client` in MLSSenderData, because that always corresponds to the `member` case.

Reply:
Good point! I'll change it as suggested.